File: src/数组、链表、跳表/Linked List/24.两两交换链表中的节点.go
/*
 * @lc app=leetcode.cn id=24 lang=golang
 *
 * [24] 两两交换链表中的节点
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
// func swapPairs(head *ListNode) *ListNode {
// 	var res *ListNode
// 	if head == nil || head.Next == nil {
//         res = head
//     } else {
//         res = head.Next
//     }
// 	pre := &ListNode{-1, head}
// 	curr := head
// 	for curr != nil && curr.Next != nil {
// 		tmp := curr.Next.Next
// 		pre.Next = curr.Next
// 		curr.Next.Next = curr
//         pre = curr
// 		curr.Next = tmp
//         curr = tmp
// 	}
// 	return res
// }
func swapPairs(head *ListNode) *ListNode {
    dummyHead := &ListNode{0, head}
    temp := dummyHead
    for temp.Next != nil && temp.Next.Next != nil {
        node1 := temp.Next
        node2 := temp.Next.Next
        temp.Next = node2
        node1.Next = node2.Next
        node2.Next = node1
        temp = node1
    }
    return dummyHead.Next
}
// @lc code=end

File: src/数组、链表、跳表/Linked List/24.两两交换链表中的节点.cs
/*
 * @lc app=leetcode.cn id=24 lang=csharp
 *
 * [24] 两两交换链表中的节点
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode SwapPairs(ListNode head) {
        var dummyHead = new ListNode() { val = -1, next = head };
        var tmp = dummyHead;
        while(tmp.next != null && tmp.next.next != null){
            var node1 = tmp.next;
            var node2 = tmp.next.next;
            tmp.next = node2;
            node1.next = node2.next;
            node2.next = node1;
            tmp = node1;
        }
        return dummyHead.next;
    }
}
// @lc code=end

